include/eve/module/complex/complex.hpp

```diff
@@ -0,0 +1,145 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/vectorizable.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/module/core.hpp>
+#include <eve/module/math.hpp>
+#include <eve/module/complex/regular/complex.hpp>
+#include <eve/module/complex/regular/detail/arithmetic.hpp>
+#include <eve/module/complex/regular/detail/math.hpp>
+#include <eve/module/complex/regular/detail/predicates.hpp>
+#include <eve/product_type.hpp>
+#include <ostream>
+
+namespace eve
+{
+  //================================================================================================
+  //! @addtogroup simd_types
+  //! @{
+  //================================================================================================
+  //! @brief SIMD-compatible representation of complex numbers
+  //!
+  //! **Required header:** `#include <eve/module/complex.hpp>`
+  //!
+  //! eve::complex is structure representing complex number and mean to be used in conjunction with
+  //! eve::wide.
+  //!
+  //! @tparam Type  Underlying floating point type
+  //================================================================================================
+  template<floating_scalar_value Type>
+  struct complex : struct_support<complex<Type>, Type, Type>
+  {
+    using eve_disable_ordering  = void;
+    using parent                = struct_support<complex<Type>, Type, Type>;
+
+    /// Underlying type
+    using value_type = Type;
+
+    /// Default constructor
+    explicit  complex(Type r = 0)     noexcept : parent{r,0} {}
+              complex(Type r, Type i) noexcept : parent{r,i} {}
+
+    /// Stream insertion operator
+    friend std::ostream& operator<<(std::ostream& os, like<complex> auto const& z)
+    {
+      return os << real(z) << std::showpos << imag(z) << "i" << std::noshowpos;
```

Comment:
I don't know how much we care, but this will break the state of os if that had showpos. Why do you need the modifier

Reply:
so I don't have to do a if to know if I need to display 3 + 4i or 3 - 4i
I can put a if instead of I need to delve into ostream flag state conservation which is also a thing.

Reply:
Wait, I suspect all of this is wrong. You will print (wide) (wide) and you are suppose to print by pairs I think.